test/test_gpu_tensorcore.cpp

```diff
@@ -3474,16 +3474,293 @@ TEST_F(NVFuserTest, FusionMatmulSegmenterBasicMatmulRelaxedCheck_CUDA) {
+  fusion->addOutput(tv3);
+
+  TORCH_CHECK(
+      1 == ir_utils::getMmaOps(fusion.get()).size(),
+      "matmul fusion must have at least one MmaOp");
+  TORCH_CHECK(
+      ir_utils::getMmaOps(fusion.get()).front()->layout().has_value(),
+      "input layout has not be set for MmaOp");
+  TORCH_CHECK(
+      MatmulLayout::TN ==
+          ir_utils::getMmaOps(fusion.get()).front()->layout().value(),
+      "input layout from test and MmaOp do not match");
+
+  FusionExecutorCache executor_cache(std::move(fusion));
+
+  const int M = 504, N = 136, K = 1024;
+
+  at::manual_seed(0);
+  at::Tensor t0 = matmulAtInput(M, N, K, layout, TensorMatmulPos::A, at::kHalf);
+  at::Tensor t1 = matmulAtInput(M, N, K, layout, TensorMatmulPos::B, at::kHalf);
+  at::Tensor t2 = atMatmul(t0.to(at::kHalf), t1.to(at::kHalf), layout);
+  at::Tensor tref = at::relu(t2).to(at::kFloat);
+
+  auto outputs = executor_cache.runFusionWithInputs({t0, t1});
+
+  TORCH_CHECK(
+      !executor_cache.getMostRecentKernelRuntime()->isSegmented(),
+      "segmentation did happen");
+
+  TORCH_CHECK(outputs[0].allclose(tref, 0.001, 0.001));
+}
+
+#if 0
+// Matmul test that relies on segmenter for 'C = relu(A x B)' fusion, for
+//   Ampere
+TEST_F(NVFuserTest, FusionMatmulSegmenterEpilogueGelu_CUDA) {
+  NVFUSER_TEST_CUDA_ARCH_RANGE_GUARD(8, 0, 8, 9);
+  const auto layout = MatmulLayout::TT;
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  // alpha - s0, A - tv 0, B - tv1
+  auto tv0 = makeContigTensor(2, DataType::Half);
+  auto tv1 = makeContigTensor(2, DataType::Half);
+
+  auto tv2 = matmul(tv0, tv1, layout, true);
+  auto tv3 = gelu(tv2);
+
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  fusion->addOutput(tv3);
+
+  TORCH_CHECK(
+      1 == ir_utils::getMmaOps(fusion.get()).size(),
+      "matmul fusion must have at least one MmaOp");
+  TORCH_CHECK(
+      ir_utils::getMmaOps(fusion.get()).front()->layout().has_value(),
+      "input layout has not be set for MmaOp");
+  TORCH_CHECK(
+      MatmulLayout::TN ==
+          ir_utils::getMmaOps(fusion.get()).front()->layout().value(),
+      "input layout from test and MmaOp do not match");
+
+  FusionExecutorCache executor_cache(std::move(fusion));
+
+  const int M = 504, N = 136, K = 1024;
+
+  at::manual_seed(0);
+  at::Tensor t0 = matmulAtInput(M, N, K, layout, TensorMatmulPos::A, at::kHalf);
+  at::Tensor t1 = matmulAtInput(M, N, K, layout, TensorMatmulPos::B, at::kHalf);
+  at::Tensor t2 = atMatmul(t0.to(at::kHalf), t1.to(at::kHalf), layout);
+  at::Tensor tref = at::gelu(t2).to(at::kFloat);
+
+  auto outputs = executor_cache.runFusionWithInputs({t0, t1});
+
+  TORCH_CHECK(
+      !executor_cache.getMostRecentKernelRuntime()->isSegmented(),
+      "segmentation did happen");
+
+  TORCH_CHECK(outputs[0].allclose(tref, 0.001, 0.001));
+}
+#endif
+
+// MMA and alpha unit test, for Ampere TN
+TEST_F(NVFuserTest, FusionAmpereMMATNAlpha_CUDA) {
+  NVFUSER_TEST_CUDA_ARCH_GUARD(8, 0);
```

Comment:
We already have `NVFUSER_TEST_CUDA_ARCH_COMPILE_CHECK` below, so I think we don't need `NVFUSER_TEST_CUDA_ARCH_GUARD`?